core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -481,8 +481,14 @@ object SparkSubmit {
     }
 
     // Let YARN know it's a pyspark app, so it distributes needed libraries.
-    if (clusterManager == YARN && args.isPython) {
-      sysProps.put("spark.yarn.isPython", "true")
+    if (clusterManager == YARN) {
+      if (args.isPython) {
+        sysProps.put("spark.yarn.isPython", "true")
+      }
+      if (args.principal != null) {
+        require(args.keytab != null, "Keytab must be specified when the keytab is specified")
```

Comment:
Actually this brings up an interesting issue - looks like the initial login would not happen if the keytab and principal are passed in only via config..Let me see how we can fix this (that should fix this issue anyway)